crates/core/src/config.rs

```diff
@@ -0,0 +1,54 @@
+use crate::error::ConfigError;
+use dirs::home_dir;
+use serde::{Deserialize, Serialize};
+
+pub const CONFIG_DIR: &str = ".pcl";
+pub const CONFIG_FILE: &str = "config.toml";
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct CliConfig {
+    pub auth: Option<UserAuth>,
+    pub assertions_for_submission: Vec<AssertionForSubmission>,
+}
+
+impl CliConfig {
+    pub fn write_to_file(&self) -> Result<(), ConfigError> {
+        let config_dir = home_dir().unwrap().join(CONFIG_DIR);
+        let config_file = config_dir.join(CONFIG_FILE);
+        let config_str = toml::to_string(self).unwrap();
+        std::fs::write(config_file, config_str).map_err(ConfigError::WriteError)?;
+        Ok(())
+    }
```

Comment:
Is it correct that this doesn't actually create the file to write to?

Reply:
it does!